common/environment/environment.cpp

```diff
@@ -1606,27 +1646,77 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByAddressPath(const char * ne
         addPathSepChar(fullDropZoneDir);
         const char * pdropzoneDir = fullDropZoneDir.str();
 
-        // Check target file path starts with this Drop zone path
-        // the drop zone paths can be nested (nothing forbids it) like
-        // dz1: /a/b/c/d
-        // dz2: /a/b/c
-        // dz3: /a/b
-        // check all and select the shortest
-
         if (strncmp(pdropzoneDir, targetFilePath, strlen(pdropzoneDir)) == 0)
         {
-            if (dropzonePathLen > strlen(pdropzoneDir))
+            dropZone = &zoneIt->query();
+
+            // For backward compatibility
+            SCMStringBuffer dropZoneMachineName;
+            // Returns dropzone '@computer' value if it is exists
+            dropZone->getComputerName(dropZoneMachineName);
+            if ( 0 != dropZoneMachineName.length())
             {
-                dropzonePathLen = strlen(pdropzoneDir);
-                dropzonePath.set(pdropzoneDir);
-                pdropzonePath = dropzonePath.str();
-                retVal = &zoneIt->query();
+                Owned<IConstMachineInfo> machineInfo = getMachine(dropZoneMachineName.str());
+                if (machineInfo)
+                {
+                    SCMStringBuffer dropZoneMachineNetAddress;
+                    machineInfo->getNetAddress(dropZoneMachineNetAddress);
+                    IpAddress dropZoneMachineIp(dropZoneMachineNetAddress.str());
+#ifdef _DEBUG
+                    StringBuffer dropZoneMachineNetAddressIpString;
+                    dropZoneMachineIp.getIpText(dropZoneMachineNetAddressIpString);
+                    LOG(MCdebugInfo, unknownJob, "Drop zone coputer: '%s', IP: '%s'", dropZoneMachineName.str(), dropZoneMachineNetAddressIpString.str());
+#endif
+                    if (targetIp.ipequals(dropZoneMachineIp))
+                       break;
+                    else
+                       // Path matched, but machine didn't
+                       dropZone = nullptr;
+                }
+
+            }
+
+            // check if it is local machine
+            //if (strcmp(netaddress, dropZoneMachineName.str()) !=  0)
```

Comment:
```
    // check if it is local machine
    //if (strcmp(netaddress, dropZoneMachineName.str()) !=  0)
```
'local' isn't correct here, these servers can be anywhere.
Maybe these are old lines from previous PR version?

Should be removed I think + newline before else